### PR TITLE
8269280: (bf) Replace StringBuffer in *Buffer.toString()

### PR DESCRIPTION
**Pull request by shipilev**
There is a StringBuffer use in X-Buffer template. The extra thread safety is not required there. In post-BiasedLocking times, this is gets worse, as every append is synchronized.

Additional testing:
 - [x] `java/nio/Buffer` tests, verified current `toString` output is unchanged

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269280](https://bugs.openjdk.java.net/browse/JDK-8269280): (bf) Replace StringBuffer in *Buffer.toString()

### Reviewers
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)
 * [Chris Hegarty](https://openjdk.java.net/census#chegar) (@ChrisHegarty - **Reviewer**)
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4581/head:pull/4581` \
`$ git checkout pull/4581`

Update a local copy of the PR: \
`$ git checkout pull/4581` \
`$ git pull https://git.openjdk.java.net/jdk pull/4581/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4581`

View PR using the GUI difftool: \
`$ git pr show -t 4581`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4581.diff">https://git.openjdk.java.net/jdk/pull/4581.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/nio/X-Buffer.java.template [hunk 1]
```diff
@@ -1666,16 +1666,11 @@ public abstract class $Type$Buffer
...
-        sb.append(limit());
-        sb.append(" cap=");
-        sb.append(capacity());
-        sb.append("]");
-        return sb.toString();
+        return getClass().getName()
+                 + "[pos=" + position()
+                 + " lim=" + limit()
+                 + " cap=" + capacity()
+                 + "]";
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/nio/X-Buffer.java.template:1673]: This looks okay. Is this a drive-by observation or is something really stressing Buffer::toString?

**Review by AlanBateman — APPROVED** [review]

File: src/java.base/share/classes/java/nio/X-Buffer.java.template (hunk 1 above)
Comment (shipilev) [reply src/java.base/share/classes/java/nio/X-Buffer.java.template:1673]: Viktor Klang made a drive-by observation here: https://twitter.com/viktorklang/status/1407688764567298052. I highly doubt it matters anywhere performance-wise, so I did no benchmarks.

**Review by ChrisHegarty — APPROVED** [review]

**Review by bplb — APPROVED** [review]

**Comment by shipilev** [comment]
Thanks for reviews!

/integrate
